[Samsung] Add missing devices

Evernow opened the pull request:
This PR attempts to add most missing popular Samsung phones of which they have readily available release information, there are however quite a lot of phones missing from this PR still, here is a list of the ones I plan to tackle (not necessarily all of them, I am looking for a way to somehow automate this):

<details>
  <summary>List of devices with approximate release dates</summary>

- [x] Galaxy Tab Active4 Pro
- [x] Galaxy A13 (SM-A137)
- [x] Galaxy Fold 5G
- [x] Galaxy Note 10+ 5G
- [x] Galaxy Note 10 5G
- [x] Galaxy Watch Active2
- [x] Galaxy Watch Active2 Aluminum
- [x] Galaxy Xcover 4s
- [x] Galaxy Watch Active
- [x] Galaxy M40
- [x] Galaxy M30
- [x] Galaxy M20
- [x] Galaxy M10
- [x] Galaxy Tab A 8.0 & S Pen (2019)
- [x] Galaxy Tab Advanced2
- [x] Galaxy Tab A 8.0 (2018)
- [x] Galaxy Tab A 10.5
- [x] Galaxy A6s
- [x] Galaxy A7 (2018)
- [x] Galaxy Watch
- [x] Galaxy On6
- [x] Galaxy J7 (2018)
- [x] Galaxy J3 (2018)
- [x] Galaxy A8 Star (A9 Star)
- [x] Galaxy S Light Luxury
- [x] Galaxy A6+ (2018)
- [x] Galaxy A6 (2018)
- [x] Galaxy J2 Pro (2018)
- [x] Galaxy A8+ (2018)
- [x] Galaxy A8 (2018)
- [x] Galaxy J2 (2017)
- [x] Galaxy C7 (2017)
- [x] Gear Sport
- [x] Galaxy J7 V
- [x] Galaxy Note FE
- [x] Galaxy J7 Max
- [x] Galaxy J7 (2017)
- [x] Galaxy J7 Pro
- [x] Galaxy J5 (2017)
- [x] Galaxy J3 (2017)
- [x] Galaxy Folder2
- [x] Z4
- [x] Galaxy S8+
- [x] Gear S3 classic LTE
- [x] Galaxy C5 Pro
- [x] Galaxy Xcover 4
- [x] Galaxy Tab S3 9.7
- [x] Galaxy J1 mini prime
- [x] Galaxy J3 Emerge
- [x] Galaxy C7 Pro
- [x] Galaxy A7 (2017)
- [x] Galaxy A5 (2017)
- [x] Galaxy A3 (2017)
- [x] Galaxy Grand Prime Plus
- [x] Galaxy J2 Prime
- [x] Galaxy C9 Pro
- [x] Galaxy A8 (2016)
- [x] Galaxy On8
- [x] Galaxy On7 (2016)
- [x] Gear S3 classic
- [x] Gear S3 frontier
- [x] Gear S3 frontier LTE
- [x] Galaxy J5 Prime
- [x] Z2
- [x] Galaxy Note 7 (USA)
- [x] Galaxy Note 7
- [x] Galaxy On7 Pro
- [x] Galaxy On5 Pro
- [x] Galaxy Tab J
- [x] Galaxy J Max
- [x] Galaxy J2 Pro (2016)
- [x] Galaxy J2 (2016)
- [x] Z3 Corporate
- [x] Galaxy Xcover 3 G389F
- [x] Galaxy S7 active
- [x] Galaxy J3 Pro
- [x] Galaxy C7
- [x] Galaxy C5
- [x] Galaxy A9 Pro (2016)
- [x] Galaxy J7 (2016)
- [x] Galaxy J5 (2016)
- [x] Galaxy Tab A 10.1 (2016)
- [x] Galaxy Tab A 7.0 (2016)
- [x] Galaxy S7 edge (USA)
- [x] Galaxy S7 (USA)
- [x] Galaxy J1 Nxt
- [x] Gear S2 classic 3G
- [x] Galaxy J1 (2016)
- [x] Galaxy A9 (2016)
- [x] Galaxy A7 (2016)
- [x] Galaxy A5 (2016)
- [x] Galaxy A3 (2016)
- [x] Galaxy Express Prime
- [x] Galaxy View
- [x] Galaxy On7
- [x] Galaxy On5
- [x] Z3
- [x] Galaxy J1 Ace
- [x] Gear S2 classic
- [x] Gear S2
- [x] Gear S2 3G
- [x] Galaxy Note 5 (USA)
- [x] Galaxy Note 5
- [x] Galaxy Note 5 Duos
- [x] Galaxy S6 edge+ (USA)
- [x] Galaxy S6 edge+
- [x] Galaxy S6 edge+ Duos
- [x] Galaxy S5 Neo
- [x] Galaxy S4 mini I9195I
- [x] Galaxy Folder
- [x] Galaxy Tab S2 9.7
- [x] Galaxy Tab S2 8.0
- [x] Galaxy A8 Duos
- [x] Galaxy A8
- [x] Galaxy V Plus
- [x] Galaxy J7 Nxt
- [x] Galaxy Tab E 9.6
- [x] Galaxy S6 active
- [x] Galaxy Tab 3 V
- [x] Galaxy Tab A 9.7 & S Pen
- [x] Galaxy Tab A 9.7
- [x] Galaxy Tab A 8.0 & S Pen (2015)
- [x] Galaxy Tab A 8.0 (2015)
- [x] Galaxy Xcover 3
- [x] Galaxy S6 edge (USA)
- [x] Galaxy S6 (USA)
- [x] Galaxy S6 edge
- [x] Galaxy S6 Duos
- [x] Galaxy S6
- [x] Galaxy J1 4G
- [x] Galaxy J2
- [x] Galaxy J1
- [x] Galaxy Tab 3 Lite 7.0 VE
- [x] Z1
- [x] Galaxy A7 Duos
- [x] Galaxy Grand Max
- [x] Galaxy E7
- [x] Galaxy E5
- [x] Galaxy Core Prime
- [x] Galaxy A5 Duos
- [x] Galaxy A5
- [x] Galaxy A3 Duos
- [x] Galaxy A3
- [x] Galaxy S5 Plus
- [x] Galaxy Pocket 2
- [x] Galaxy V
- [x] Galaxy Grand Prime Duos TV
- [x] Galaxy Grand Prime
- [x] Galaxy Ace Style LTE G357
- [x] Galaxy Note Edge
- [x] Galaxy Note 4 Duos
- [x] Galaxy Note 4
- [x] Galaxy Tab Active LTE
- [x] Galaxy Tab Active
- [x] Galaxy Mega 2
- [x] Gear S
- [x] Gear 2 Neo
- [x] Gear Live
- [x] Gear 2
- [x] Galaxy Gear
- [x] Galaxy S5 LTE-A G901F
- [x] Galaxy Alpha (S801)
- [x] Galaxy Alpha
- [x] Galaxy S5 mini Duos
- [x] Galaxy Avant
- [x] Galaxy S Duos 3
- [x] Galaxy Ace NXT
- [x] Galaxy Star 2 Plus
- [x] Galaxy S5 mini
- [x] Galaxy Ace 4 LTE G313
- [x] Galaxy Ace 4
- [x] Galaxy Young 2
- [x] Galaxy Star 2
- [x] Galaxy Core II
- [x] Galaxy S5 Sport
- [x] Galaxy S5 LTE-A G906S
- [x] Galaxy Tab S 8.4 LTE
- [x] Galaxy Tab S 8.4
- [x] Galaxy Tab S 10.5 LTE
- [x] Galaxy Tab S 10.5
- [x] Galaxy Core Lite LTE
- [x] I9301I Galaxy S3 Neo
- [x] Galaxy W
- [x] Galaxy S5 Active
- [x] Galaxy K zoom
- [x] Galaxy Beam2
- [x] I9300I Galaxy S3 Neo
- [x] Galaxy Ace Style
- [x] ATIV SE
- [x] Galaxy Tab 4 7.0
- [x] Galaxy Tab 4 7.0 3G
- [x] Galaxy Tab 4 7.0 LTE
- [x] Galaxy Tab 4 8.0
- [x] Galaxy Tab 4 8.0 3G
- [x] Galaxy Tab 4 8.0 LTE
- [x] Galaxy Tab 4 10.1
- [x] Galaxy Tab 4 10.1 3G
- [x] Galaxy Tab 4 10.1 LTE
- [x] G3812B Galaxy S3 Slim
- [x] I8200 Galaxy S III mini VE
- [x] Galaxy S5 Duos
- [x] Galaxy S5 (octa-core)
- [x] Galaxy S5 (USA)
- [x] Galaxy S5
- [x] Galaxy Core LTE G386W
- [x] Galaxy Core LTE
- [x] Galaxy Star Trios S5283
- [x] Galaxy Note 3 Neo Duos
- [x] Galaxy Note 3 Neo
- [x] Galaxy Tab 3 Lite 7.0 3G
- [x] Galaxy Tab 3 Lite 7.0
- [x] Galaxy Grand Neo
- [x] Galaxy Note Pro 12.2 LTE
- [x] Galaxy Note Pro 12.2 3G
- [x] Galaxy Note Pro 12.2
- [x] Galaxy Tab Pro 12.2 LTE
- [x] Galaxy Tab Pro 12.2 3G
- [x] Galaxy Tab Pro 12.2
</details>

There's over a hundred TODOs, the majority are approximate release dates which are fine (these were parsed from various sources, but exact day could not be had, rest are mostly just no exact end of life date given.

Evernow committed:
[Samsung] Add missing devices

Evernow committed:
Add almost all remaining devices

Evernow committed:
Get rid of duplicate Galaxy S5

Evernow committed:
Fix some support status that are under quarterly

Evernow committed:
Fix some support status for devices released last year

Evernow committed:
Update samsungmobile.md

Evernow committed:
Fix mistakes with Galaxy A13

adriens reviewed (commented):
Will you also add the "Galaxy S23 Ultra" ?
https://www.samsung.com/fr/smartphones/galaxy-s23-ultra/

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -11,14 +11,119 @@ releaseColumn: false
+-   releaseCycle: "Galaxy Tab Active 4 Pro"
+    support: true
+    eol: false # https://doc.samsungmobile.com/SM-T636B/XSA/doc.html
+    releaseDate: 2022-09-13 # TODO: Verify date
```
LGTM:  availability vary, but it was september in Europe : https://news.samsung.com/global/introducing-the-galaxy-tab-active4-pro-a-ruggedized-device-designed-for-the-new-mobile-workforce.

Evernow replied:
Perhaps we should add a note at the bottom that the release date mentioned may vary depending on the region?

marcwrobel replied:
Yep, that would be a nice clarification. I opened #2463 to not forget it (if possible I would like no more modification in this PR).

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -30,27 +135,112 @@ releases:
+    releaseDate: 2022-02-25
 
+-   releaseCycle: "Galaxy Tab S8 Ultra"
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Only 6 months of support !?

Same remark for the following models such as Galaxy Tab S8+.

Evernow replied:
Looking into this, I think in a later PR we really need to redefine the full support column, but for now I think if a device has received android 12 we mark it as fully supported, and then potentially revisit this with android 13 and mark those that didn't get android 13 as not fully supported anymore. 

This approach would mark the two Tab S8 as fully supported for now, do you agree with this?

marcwrobel replied:
Yes.

FYI according to https://www.sammobile.com/news/every-samsung-galaxy-device-updated-to-android-13-one-ui-5-0/ this model is supporting Android 13.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -30,27 +135,112 @@ releases:
+    eol: false # https://doc.samsungmobile.com/SM-A035G/BTU/doc.html
+    releaseDate: 2022-01-21
+    
+-   releaseCycle: "Galaxy A03 Core"
```
This comment is just a marker to indicate I did the review of the previous models. Will continue later, do not remove.

marcwrobel replied:
For the record I used https://www.gsmarena.com/ for the review. @Evernow do you think it could be a good thing to link to this site ?

Evernow replied:
> For the record I used https://www.gsmarena.com/ for the review. @Evernow do you think it could be a good thing to link to this site ?

If you mean link to the site in the .md as a comment, I don't see anything wrong with that, may be a lot of comments though, I don't think it's strictly necessary unless the release date is a bit iffy

marcwrobel replied:
I meant using the link property to point to https://www.gsmarena.com/. It is unusual for endoflife.date but documentation on this website looks way better than Samsung's own documentation.

Evernow committed:
Rename Samsung W21 for consistency

Caught by @adriens

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -30,27 +135,112 @@ releases:
+    releaseDate: 2022-04-14
+
+-   releaseCycle: "Galaxy Tab S8"
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Not sure about this one, according to https://www.gsmarena.com/samsung_galaxy_tab_s8-11343.php this model is upgradable to Android 13.

marcwrobel replied:
Confirmed on https://www.sammobile.com/news/every-samsung-galaxy-device-updated-to-android-13-one-ui-5-0/.

This is not the only one that should not have been marked out of support, this is also the case for Galaxy F42 5G, Galaxy M52 5G, Galaxy M22...

marcwrobel replied:
I wrote a quick script (committed in this PR, can be removed later) that check if devices has a support date and are listed in https://www.sammobile.com/news/every-samsung-galaxy-device-updated-to-android-13-one-ui-5-0/. I think those devices should be marked as supported :

```
$ python3 _auto/samsung.py 
Galaxy F42 5G (2022-10-01)
Galaxy M52 5G (2022-10-01)
Galaxy M22 (2022-10-01)
Galaxy M32 5G (2022-10-01)
Galaxy A03s (2022-10-01)
Galaxy F22 (2022-10-01)
Galaxy M32 (2022-10-01)
Galaxy A22 5G (2022-10-01)
Galaxy A22 (2022-10-01)
Galaxy Tab S7 FE (2022-10-01)
Galaxy M42 5G (2022-10-01)
Galaxy M62 (2022-10-01)
Galaxy F62 (2022-10-01)
Galaxy A32 5G (2022-10-01)
Galaxy M12 (2022-10-01)
Galaxy Tab S7 (2022-10-01)
Galaxy Tab S6 Lite (2022-10-01)
Galaxy S10 Lite (2022-10-01)
```

Evernow replied:
I agree with you, I believe this was a misconception in my part, we cannot assume a device is no longer supported if it just received the latest android version, we can only assume a device is no longer supported if it doesn't receive the next android version.

marcwrobel replied:
Also checked products marked as unsupported less than one year after their release date, got :

```
$ python3 _auto/samsung.py 
Galaxy A13 (192)
Galaxy Tab S8 Ultra (154)
Galaxy Tab S8+ (170)
Galaxy Tab S8 (193)
Galaxy Tab A8 10.5 (2021) (257)
Galaxy A03 (253)
Galaxy A03 Core (299)
```

All those are still supported according to https://www.sammobile.com/news/every-samsung-galaxy-device-updated-to-android-13-one-ui-5-0/ or https://www.gsmarena.com/.

The script is :

```python
import frontmatter
from datetime import date

with open("products/samsungmobile.md", 'r') as f:
    product = frontmatter.load(f)
    for cycle in product['releases']:
        name = cycle['releaseCycle']
        support = cycle['support']
        release = cycle['releaseDate']
        if isinstance(support, date) and ((support - release).days < 365):
            print(f"{name} ({(support - release).days})")
```

marcwrobel replied:
Models updated, resolving this conversation.

marcwrobel committed:
Add script to check samsung devices Android 13 support

marcwrobel committed:
[samsungmobile] Marks some models as still supported

According to https://www.gsmarena.com/ and https://www.sammobile.com/news/every-samsung-galaxy-device-updated-to-android-13-one-ui-5-0/.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -61,18 +251,48 @@ releases:
+
 -   releaseCycle: "Galaxy M21 2021"
-    support: false
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
According to https://www.gsmarena.com/samsung_galaxy_m21_2021-11015.php this one is still on Android 11 with no possibilities to upgrade to Android 12 or 13. So LGTM, but maybe we should update the comment and better document the rule ?

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -96,8 +316,13 @@ releases:
     eol: false
     releaseDate: 2021-06-03
 
+-   releaseCycle: "Galaxy F52 5G"
```
Same as Galaxy M21 2021

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -96,8 +316,13 @@ releases:
+
 -   releaseCycle: "Galaxy Tab A7 Lite"
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
This one is upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_tab_a7_lite-10933.php. Should we consider it supported ?

marcwrobel replied:
> I think the biggest issue is the full support section, if you agree with my comment in one of your comments, I'll change tit o state whenever applicable that a device that has received android 12 is fully supported for now, and then we change the date appropriately when we see if it receives the next major android update.

Sounds like a good rule to start. Will document the rule and fix the model if they are compatible with Android 12.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -107,7 +332,7 @@ releases:
     releaseDate: 2021-05-25
 
 -   releaseCycle: "Galaxy A82 5G"
```
Note: This one has two name : https://www.gsmarena.com/samsung_accidentally_confirms_the_galaxy_a82_5g_existence-news-48968.php.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -122,7 +347,7 @@ releases:
 
 -   releaseCycle: "Galaxy F12"
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Upgradable to Android 13 according to https://www.gsmarena.com/samsung_galaxy_f12-10825.php. Should be marked supported.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -137,10 +362,15 @@ releases:
 
 -   releaseCycle: "Galaxy A72"
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_a72-10469.php. Should we consider it supported ?

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -157,7 +387,7 @@ releases:
 
 -   releaseCycle: "Galaxy A32"
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Upgradable to Android 13 according to https://www.gsmarena.com/samsung_galaxy_a32-10753.php. Should be marked supported.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -177,7 +407,7 @@ releases:
 
 -   releaseCycle: "Galaxy A02" #This loses active support with Android 12.
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Small marker for me, do not remove.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -246,8 +476,18 @@ releases:
+    eol: false # https://doc.samsungmobile.com/SM-G780G/BTU/doc.html
+    releaseDate: 2020-10-02
+
 -   releaseCycle: "Galaxy Tab Active 3" # Confirm this.
```
Should be `Active3`. See https://www.gsmarena.com/samsung_galaxy_tab_active3-10476.php.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -246,8 +476,18 @@ releases:
+
 -   releaseCycle: "Galaxy Tab Active 3" # Confirm this.
-    support: true #This loses active support with Android 12.
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
This model is upgradable to Android 12, see https://www.gsmarena.com/samsung_galaxy_tab_active3-10476.php.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -256,16 +496,36 @@ releases:
     releaseDate: 2020-09-18
 
--   releaseCycle: "Galaxy Tab A7" #This loses active support with Android 12.
+-   releaseCycle: "Galaxy Tab A7 10.4 (2020)"
```
Note : there is also a 2022 version https://www.gsmarena.com/samsung_galaxy_tab_a7_10_4_(2022)-11988.php.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -256,16 +496,36 @@ releases:
+    eol: false # https://doc.samsungmobile.com/SM-N981U1/VZW/doc.html
+    releaseDate: 2020-08-21
+
+-   releaseCycle: "Galaxy A51 5G UW"
```
Still on android 10 according to https://www.gsmarena.com/samsung_galaxy_a51_5g_uw-10371.php, should be marked unsupported ?

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -286,21 +546,41 @@ releases:
     eol: false
     releaseDate: 2020-08-06
 
+-   releaseCycle: "Galaxy Watch3"
```
Watch3 is running on Tizen OS 5.5. We should consider documenting EOL for this OS.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -286,21 +546,41 @@ releases:
 
 -   releaseCycle: "Galaxy Tab S7+"
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Upgradable to Android 13 according to https://www.gsmarena.com/samsung_galaxy_tab_s7+-10336.php, should be marked as supported.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -286,21 +546,41 @@ releases:
     eol: false
     releaseDate: 2020-08-05
 
 -   releaseCycle: "Galaxy Tab S7 "
```
Duplicate of `"Galaxy Tab S7"` ?

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -286,21 +546,41 @@ releases:
     eol: false
     releaseDate: 2020-08-05
 
+-   releaseCycle: "Galaxy M01 Core"
```
Maybe we should add some note on the Android page about Android Go (https://www.android.com/intl/en_ph/versions/go-edition/).

marcwrobel replied:
#2731

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -286,21 +546,41 @@ releases:
+    eol: false # https://doc.samsungmobile.com/SM-M013F/INS/doc.html
+    releaseDate: 2020-07-29
+
+-   releaseCycle: "Galaxy M01s"
```
Only upgradable to Android 10, should be marked unsupported ?

marcwrobel replied:
Let's leave it like that for the time being.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -286,21 +546,41 @@ releases:
+    eol: false # https://doc.samsungmobile.com/SM-M017F/INS/doc.html
+    releaseDate: 2020-07-16
+
+-   releaseCycle: "Galaxy A71 5G UW"
```
Only Android 10 is available, should be marked unsupported ?

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -311,6 +591,11 @@ releases:
     eol: false
     releaseDate: 2020-06-15
 
+-   releaseCycle: "Galaxy S20 5G UW"
```
Only Android 10 is available, should be marked unsupported ?

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -336,18 +621,33 @@ releases:
     eol: false # Quarterly
     releaseDate: 2020-04-27
 
+-   releaseCycle: "Galaxy J2 Core (2020)"
```
Is Android 8.1 Oreo (Go edition) still supported (see https://www.gsmarena.com/samsung_galaxy_j2_core_(2020)-10208.php) ?

marcwrobel replied:
Nope, according to https://en.wikipedia.org/wiki/Android_Go. Will mark this model as unsupported.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -336,18 +621,33 @@ releases:
+    releaseDate: 2020-04-27
+
+-   releaseCycle: "Galaxy Xcover FieldPro"
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
Only on Android 8.0 (Oreo), earlier date may be more appropriate.

marcwrobel replied:
Last update on 2022-08 (https://doc.samsungmobile.com/SM-G889YB/DBT/doc.html), keeping the date.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -432,12 +747,17 @@ releases:
 
--   releaseCycle: "Galaxy Fold" #This loses active support with Android 12.
+-   releaseCycle: "Galaxy A30s"
     support: true
```
Only upgradable to Android 11, should be marked as unsupported ?

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -432,12 +747,17 @@ releases:
+    releaseDate: 2019-09-11
+
+-   releaseCycle: "Galaxy Fold"
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_fold-9523.php.

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -461,6 +781,21 @@ releases:
     releaseDate: 2019-09-01
 
+-   releaseCycle: "Galaxy Fold 5G"
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_fold_5g-9851.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -472,12 +807,12 @@ releases:
     releaseDate: 2019-08-27
 
 -   releaseCycle: "Galaxy Note 10+" #This loses active support with Android 12.
```
Note10+ according to https://www.gsmarena.com/samsung_galaxy_note10+-9732.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -472,12 +807,12 @@ releases:
 
 -   releaseCycle: "Galaxy Note 10+" #This loses active support with Android 12.
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_note10+-9732.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -472,12 +807,12 @@ releases:
     eol: false
     releaseDate: 2019-08-23
 
 -   releaseCycle: "Galaxy Note 10" #This loses active support with Android 12.
```
Note10 according to https://www.gsmarena.com/samsung_galaxy_note10-9788.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -472,12 +807,12 @@ releases:
 
 -   releaseCycle: "Galaxy Note 10" #This loses active support with Android 12.
-    support: true
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_note10-9788.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -486,16 +821,36 @@ releases:
     eol: false
     releaseDate: 2019-08-01
 
+-   releaseCycle: "Galaxy Note 10+ 5G"
```
Note10+ according to https://www.gsmarena.com/samsung_galaxy_note10+_5g-9787.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -486,16 +821,36 @@ releases:
     releaseDate: 2019-08-01
 
+-   releaseCycle: "Galaxy Note 10+ 5G"
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_note10+_5g-9787.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -486,16 +821,36 @@ releases:
+    eol: false # https://doc.samsungmobile.com/SM-N976U/TMB/doc.html
+    releaseDate: 2019-08-01 # Approximate to the month and year.
+
+-   releaseCycle: "Galaxy Note 10 5G"
```
Note10 according to https://www.gsmarena.com/samsung_galaxy_note10_5g-9789.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -486,16 +821,36 @@ releases:
+    releaseDate: 2019-08-01 # Approximate to the month and year.
+
+-   releaseCycle: "Galaxy Note 10 5G"
+    support: 2022-10-01 # Approximate from here: https://9to5google.com/2022/10/04/samsung-android-12-update-rollout/
```
upgradable to Android 12 according to https://www.gsmarena.com/samsung_galaxy_note10_5g-9789.php

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -517,10 +872,20 @@ releases:
+    eol: false # https://doc.samsungmobile.com/SM-P205/XXV/doc.html
+    releaseDate: 2019-04-01 # Approximate to the month and year.
+
 -   releaseCycle: "Galaxy View2"
```
Marker for my review

I would like to automate a little bit things with a custom field linking to https://www.gsmarena.com/. We must ask them first, because [their terms of use](https://www.gsmarena.com/terms.php3) states that :

> USE OF WEBSITE INFORMATION
> 
> You may download, view, copy and print documents and graphics incorporated in these documents (the "Documents") > from this website subject to the following: (1) the Documents may be used solely for personal, informational, non-commercial purposes; and (2) the Documents may not be modified or altered in any way. Except as expressly provided herein, you may not use, download, upload, copy, print, display, perform, reproduce, publish, license, post, transmit or distribute any information from this website in whole or in part without the prior written permission of GSMArena.com.

@endoflife-date/everyone do you think it's a good idea ?

marcwrobel replied:
Rate limiting is too much sensitive to automate anything with gsmarena.

Found another source of data on https://github.com/ilyasozkurt/mobilephone-brands-and-models but it was not updated since 10/09/2021. For the record :

```
http https://raw.githubusercontent.com/ilyasozkurt/mobilephone-brands-and-models/master/devices.json | jq '.RECORDS[] | select(.name | test("^Samsung.+", "i")) | {name: .name, released_at: .released_at, os: .os}'
```

marcwrobel commented on products/samsungmobile.md:
```diff
@@ -721,10 +1171,960 @@ releases:
+    eol: 2019-02-01 # https://doc.samsungmobile.com/SM-A910F/INS/doc.html
+    releaseDate: 2016-05-01 # Approximate to the month and year.
+
+-   releaseCycle: "Galaxy Tab A 10.1 (2016)"
```
Marker for my review.

Pfiou, I do not know how you did to add so much model @Evernow ;)

marcwrobel committed:
fixes

marcwrobel committed:
fix

marcwrobel reviewed (approved):
LGTM, merging.

I must admit I did not check most of the models without link to doc.samsungmobile.com prior to 2016-04, but I doubt we can find much information anyway.

I think this page could be improved by :

- using the dump on https://github.com/ilyasozkurt/mobilephone-brands-and-models (see https://github.com/endoflife-date/endoflife.date/pull/2387/files#r1144996311) to verify all old models and enrich the data (by retrieving release date, the model number...)
- moving links to doc.samsungmobile.com to the link attribute to open the way for some automation (we could extract the latest update dates and guess the EOL date).
- use https://security.samsungmobile.com/workScope.smsb to automatically check for new models / models without support anymore (EOL ?).
- fix the description : it says there are four support brackets, but only describe three.

I also think we should drop the support column : despite the time I spent on this page I could not come up with any rule to decide what to put in this field. It could also make sense to move `eol` to `discontinued` (considering this page is about hardware and there seems to be no real policy there on the Android support).

Let me know @Evernow if some of these ideas looks interesting.
